File: Android/2016-04-05-N-Preview-MultiWindow/app/src/main/java/tech/thdev/multiwindow/base/presenter/BasePresenter.java
package tech.thdev.multiwindow.base.presenter;


import tech.thdev.multiwindow.base.presenter.view.BasePresenterView;

/**
 * Created by Tae-hwan on 4/26/16.
 */
public class BasePresenter<V extends BasePresenterView> {

    private V view;

    public BasePresenter(V view) {
        this.view = view;
    }

    protected V getView() {
        return view;
    }
}

File: Android/2016-07-21-kotlin-example-01/app/src/main/kotlin/tech/thdev/kotlin_example_01/base/presenter/BasePresenter.kt
package tech.thdev.kotlin_example_01.base.presenter

/**
 * Created by Tae-hwan on 7/21/16.
 */
interface BasePresenter<in VIEW> {

    /**
     * View Attach.
     */
    fun attachView(view: VIEW)

    /**
     * View detach
     */
    fun detachView()
}